### ISSUE TITLE
First Dozer/Worker spawn at random positions

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#2427]
First Dozer/Worker spawn at random positions.

Do they need to?

### Match 1

![shot_20240618_135013_27](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/cdacd7c2-0747-4128-a2a5-5421169ce785)

### Match 2

![shot_20240618_135020_28](https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/1fc6968c-8a7e-46f5-834e-5b53e6f50d1a)

**Comment by penfriendz** [comment]
Good issue.

I think dozers/workers should spawn consistently. Ideally I would want them to spawn near the cc but at a position as close as possible to the nearest supply.

**Comment by MTKing4** [comment]
No.

Streamline the positions, should be the same for all factions as well, something like at the center of the south west edge of geometry of the CC.

**Comment by penfriendz** [comment]
No. There should be no advantage for spawning in top, bottom, left, or right position or anywhere in between. This is one of the biggest spawning asymmetries and should be fixed if possible.

**Comment by FiReArM-ZH** [comment]
> No. There should be no advantage for spawning in top, bottom, left, or right position or anywhere in between. This is one of the biggest spawning asymmetries and should be fixed if possible.

Even now dozer/worker don't spawn north of the CC.

I think it's best if dozer/worker spawns right in front of door

**Comment by penfriendz** [comment]
> > No. There should be no advantage for spawning in top, bottom, left, or right position or anywhere in between. This is one of the biggest spawning asymmetries and should be fixed if possible.
> 
> Even now dozer/worker don't spawn north of the CC.
> 
> I think it's best if dozer/worker spawns right in front of door

I know it doesn't spawn north now. That's not a reason as to why it shouldn't be able to spawn anywhere which is most fair.

How about having it spawn in the middle of the CC, or where it normally gets built, and then the player can move it any which direction just like a freshly built unit?

**Comment by xezon** [comment]
> I think it's best if dozer/worker spawns right in front of door

I agree with this. It is conceptually the simplest. As long as it is consistent, it puts players at even playing field when playing reversed matchup + spots.

**Comment by MTKing4** [comment]
> > No. There should be no advantage for spawning in top, bottom, left, or right position or anywhere in between. This is one of the biggest spawning asymmetries and should be fixed if possible.
> 
> Even now dozer/worker don't spawn north of the CC.
> 
> I think it's best if dozer/worker spawns right in front of door

I thought about this too but it wouldn't be 100% equal because GLA door is placed differently than China/USA ones being at the center as opposed to the right

**Comment by MTKing4** [comment]
> > I think it's best if dozer/worker spawns right in front of door
> 
> I agree with this. It is conceptually the simplest. As long as it is consistent, it puts players at even playing field when playing reversed matchup + spots.

It would not be consistent because GLA door is in the middle of the CC unlike China/USA which is on the right

That's why i suggested it to be at center of the south west edge of of the geometry of all CCs, this way it's 100% symmetrical across all 3 factions because they have an equal sized geometry

**Comment by MTKing4** [comment]
As for north Vs south maps, it's best to flip the CCs upside down, which should be done with thyme i reckon, and then everything will be fair
